src/index.tsx

```diff
@@ -73,11 +72,29 @@ var createWebmarkFolder = (): void => {
     )
 };
 
-var getCurrentUrl = (): string => {
-    return 'https://current.url.com';
+// type checker for proper string type url
+let typeCheckString = (text: string | undefined): string => {
+    if (text != undefined) {
+        return text;
+    } else {
+        // TODO: this block needs proper type guard rather than returning below url.
+        return "https://example.com/error";
+    }
 };
 
-var saveToWebmarkFolder = (url: string): void => {
+let getCurrentUrlAndSave = () => {
+    chrome.tabs.query(
+        { active: true, currentWindow: true },
+        ([currentTab]) => {
+            const thisTabUrl: string = typeCheckString(currentTab.url);
+            const thisTabTitle: string = typeCheckString(currentTab.title);
+            saveToWebmarkFolder(thisTabUrl, thisTabTitle);
+        });
+}
+
+
+
+var saveToWebmarkFolder = (url: string, title: string): void => {
```

Comment:
I would take `url: string | undefined, title: string | undefined)` as parameters and take care that here (without creating another function to clean it). What do you think?

Reply:
because I did same operation twice, I created a separate function to handle it. 
speaking of that, I wonder what would happen when saving a page without <title> tag. Would it be undefined? Or would it be an empty string?

Reply:
1. The two operations both take in a `string | undefined` and return `string` but return a different value when it takes in an `undefined`. If so, I think they're two different functions.
2. I assume it would be `undefined`, but we'll have to see it ourselves. Can't find on Google search results.

Reply:
1. Hmm, you are right. But wouldn't it make `saveToWebmarkFolder` a one huge function? If that is fine I see your point .